### PR TITLE
eir: Implement Multiboot2

### PR DESCRIPTION
**Pull request by thomtl**

### CODE REVIEW

File: kernel/eir/multiboot2/multiboot2.cpp [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+	kMb2TagEfiBs = 18,
+	kMb2TagEfi32ImageHandle = 19,
+	kMb2TagEfi64ImageHandle = 20,
+	kMb2TagLoadBaseAddr = 21
+};
+
+extern "C" void eirMultiboot2Main(uint32_t info, uint32_t magic){
+	if(magic != 0x36d76289){
+		frigg::infoLogger() << "eir: Invalid multiboot2 signature, halting..." << frigg::endLog;
+		asm("cli; hlt");
```
Comment (avdgrinten) [inline kernel/eir/multiboot2/multiboot2.cpp]: `cli; hlt` can be exited by a NMI. Use the `panicLogger` instead.

File: kernel/eir/multiboot2/multiboot2.cpp [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+	Mb2TagFramebuffer* framebuffer = nullptr;
+
+	uintptr_t mmap_start = 0;
+	uintptr_t mmap_end = 0;
+
+	size_t n_modules = 0;
+
+	uintptr_t kernel_module_start = 0;
+	
+	frigg::StringView cmdline{};
```
Comment (avdgrinten) [inline kernel/eir/multiboot2/multiboot2.cpp]: We should prefer `frg` over `frigg` for new code.

File: kernel/eir/multiboot2/multiboot2.cpp [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+	for(size_t i = 8 /* Skip size and reserved fields*/; i < mb_info->size; i += add_size){
+		Mb2Tag* tag = (Mb2Tag*)((uint8_t*)info + i);
+
+		if(tag->type == kMb2TagEnd)
+			break;
+
+		add_size = tag->size;
+		if((add_size % 8)!= 0) add_size += (8 - add_size % 8); // Align 8byte
```
Comment (avdgrinten) [inline kernel/eir/multiboot2/multiboot2.cpp]: Coding style: `if(add_size % 8)`; put the statement on its own line.

File: kernel/eir/multiboot2/multiboot2.cpp [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+			}
+		}
+	}
+
+	bootMemoryLimit = (bootMemoryLimit + address_t(kPageSize - 1))
+			& ~address_t(kPageSize - 1);
+
+	initProcessorEarly();
+
+	assert(mmap_start != 0);
```
Comment (avdgrinten) [inline kernel/eir/multiboot2/multiboot2.cpp]: `assert(mmap_start);`

File: kernel/eir/multiboot2/multiboot2.cpp (hunk 2 above)
Comment (thomtl) [reply kernel/eir/multiboot2/multiboot2.cpp]: `frg` namespace doesn't seem to exist?

**Pull request merged**